apps/dot_ext/oauth2_backends.py

```diff
@@ -16,22 +16,22 @@ def create_token_response(self, request):
         SMART on FHIR Authorization
         http://docs.smarthealthit.org/authorization/
         """
-        uri, http_method, body, headers = self._extract_params(request)
-        extra_credentials = self._get_extra_credentials(request)
 
-        headers, body, status = self.server.create_token_response(uri, http_method, body,
-                                                                  headers, extra_credentials)
-        fhir_body = json.loads(body)
-        token = AccessToken.objects.get(token=fhir_body["access_token"])
+        uri, headers, body, status = super(OAuthLibSMARTonFHIR, self).create_token_response(request)
 
-        if Crosswalk.objects.filter(user=token.user).exists():
+        # cribed from
+        # https://github.com/evonove/django-oauth-toolkit/blob/2cd1f0dccadb8e74919a059d9b4985f9ecb1d59f/oauth2_provider/views/base.py#L192
+        if status == 200:
             fhir_body = json.loads(body)
-            cw = Crosswalk.objects.get(user=token.user)
-            fhir_body["patient"] = cw.fhir_id
-            body = json.dumps(fhir_body)
-        uri = headers.get("Location", None)
-        if token.user.email:
-            # If email is not blank, then send an email notifications.
-            send_access_token_notifcation(token)
+            token = AccessToken.objects.get(token=fhir_body.get("access_token"))
+
+            if Crosswalk.objects.filter(user=token.user).exists():
+                fhir_body = json.loads(body)
+                cw = Crosswalk.objects.get(user=token.user)
+                fhir_body["patient"] = cw.fhir_id
+                body = json.dumps(fhir_body)
+            if token.user.email:
+                # If email is not blank, then send an email notifications.
+                send_access_token_notifcation(token)
```

Comment:
This is a little off topic, but can you remove this email while you're here? We'll re-add it later when we have the chance to be a little more comprehensive in our solution

Reply:
done